perception/image_projection_based_fusion/src/roi_cluster_fusion/node.cpp

```diff
@@ -274,11 +277,15 @@ bool RoiClusterFusionNode::out_of_scope(const DetectedObjectWithFeature & obj)
   return is_out;
 }
 
-bool RoiClusterFusionNode::filter_by_distance(const DetectedObjectWithFeature & obj)
+double RoiClusterFusionNode::get_object_square_distance(const DetectedObjectWithFeature & obj)
 {
   const auto & position = obj.object.kinematics.pose_with_covariance.pose.position;
-  const auto square_distance = position.x * position.x + position.y + position.y;
-  return square_distance > trust_distance_ * trust_distance_;
+  return position.x * position.x + position.y * position.y;
+}
+
+bool RoiClusterFusionNode::filter_by_distance(const DetectedObjectWithFeature & obj)
+{
+  return get_object_square_distance(obj) > trust_distance_ * trust_distance_;
```

Comment:
[Q] Does it need to calculate **square** distance? I feel it is enough to calculate distance like following.
```cpp
double RoiClusterFuisionNode::get_object_distance(const DetectedObjectWithFeature & obj)
{
  const auto & position = obj.object.kinematics.pose_with_covariance.pose.position;
  return std::hypot(position.x, position.y);
}
```
[IMO] Also, `filter_by_distance` sounds ambiguous, so how about `is_far_enough` for example?
The following member function allow us to apply both `trust_distance_` and `iou_x_use_distance_threshold_` with single function.
```cpp
bool RoiClusterFusionNode::is_far_enough(const DetectedObjectWithFeature & obj, const double threshold)
{
  return get_object_distance(obj) > threshold;
}
```

Reply:
>[Q] Does it need to calculate square distance? I feel it is enough to calculate distance like following.

I think squaring is fine for computational savings.


>[IMO] Also, filter_by_distance sounds ambiguous, so how about is_far_enough for example?
The following member function allows us to apply both trust_distance_ and iou_x_use_distance_threshold_ with a single function.

I agree with you :+1:

Reply:
@ktro2828 @yukkysaito Thank you for your comments and suggestion. 
> I think squaring is fine for computational savings.

So I keep using it change function `is_far_enough` for commonly use in  https://github.com/autowarefoundation/autoware.universe/pull/5148/commits/3ae96c0a0e82f01daec611776a70e089f497cc80